simulaqron/cqc_backend/cqcMessageHandler.py

```diff
@@ -399,6 +399,10 @@ def cmd_measure(self, cqc_header, cmd, xtra, inplace=False):
             return False
 
         logging.debug("CQC %s: Measured outcome %d", self.name, outcome)
+
+        # Store the outcome in self.references under the reference specified in the CQCAssignHeader
+        self.references[cqc_header.app_id][xtra.ref_id] = outcome
```

Comment:
Is this quaranteed to contain the key `xtra.ref_id`? If not, consider using a `defaultdict` as I also mentioned in the PR of CQC-Python

Reply:
I have changed the `dict` to a `defaultdict` as you mentioned in the PR of CQC-Python. However, this line stays the same. Usually, the key `xtra.ref_id` is not yet present in the dictionary, which is why it is created in this line and assigned the value `outcome`. 
If I misunderstood or made a mistake, can you clarify what needs to change?

Reply:
I misstyped and had meant the key `cqc_header.app_id`, but since that's a `defaultdict` now that's fine :)